frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -100,16 +109,16 @@ local footerTextGeneratorMap = {
             local rss = infos:match("^%S+ (%S+) ")
             -- we got the nb of 4Kb-pages used, that we convert to Mb
             rss = math.floor(tonumber(rss) * 4096 / 1024 / 1024)
-            return ("M:%d"):format(rss)
+            return ("⌨ %d"):format(rss)
```

Comment:
That's a keyboard, right? Do we have a chip-like glyph available instead? That vaguely sounds more appropriate her,e but I'm nitpicking ;).

Reply:
A quick look at FreeSerif and the Unicode listing would point to "no". Unless "tape drive" (✇ U+2707) counts ;p

Reply:
Or use these ones, swithing from the non-distracting white one to the heavily distracting black one when memory reaches some level:
![image](https://user-images.githubusercontent.com/24273478/63001251-7c73ef80-be73-11e9-9fbb-24ac5944be3e.png)

At 200M on my Kobo, things begin to get slow - but I don't know how to decide that triggering limit scientifically :)

Reply:
> At 200M on my Kobo, things begin to get slow - but I don't know how to decide that triggering limit scientifically :)

You won't reach that much memory unless it's a large and/or complex document, so that's probably expected for page turning responsiveness, but not so much for menu responsiveness.

You can do it quote unquote scientifically by simply filling up a couple 100 MB on the side. An interesting suggestion that might work on busybox (depending on how much Bash syntax it supports):

```sh
function malloc() {
  if [[ $# -eq 0 || $1 -eq '-h' || $1 -lt 0 ]] ; then
    echo -e "usage: malloc N\n\nAllocate N mb, wait, then release it."
  else 
    N=$(free -m | grep Mem: | awk '{print int($2/10)}')
    if [[ $N -gt $1 ]] ;then 
      N=$1
    fi
    sh -c "MEMBLOB=\$(dd if=/dev/urandom bs=1MB count=$N) ; sleep 1"
  fi
}
```
From https://unix.stackexchange.com/a/99390

Reply:
> You won't reach that much memory unless it's a large and/or complex document

We reach it very easily when using dict or wikipedia lookups. The TextBoxWidget code uses a lot of memory for the various Lua tables holding chars/widths/justification.
Do that, switch EPUB documents, do that again, and the memory might end up very fragmented and hardly re-used, and the usage rises.
At some point, using a gesture to show the History or TOC or top menu gets from instant to 1 or 2 seconds delay.

Reply:
Does the problem go away after a little while (i.e., post garbage collection)? Perhaps it could be of interest to trigger garbage collection as the final step in closing a document. On the flipside, it'd probably make it slower to reopen.

Reply:
Nope. I may just sometimes get -1 or -2 Mb when switching to another EPUB document, after the next page turn.

> trigger garbage collection as the final step in closing a document

We already do that:
https://github.com/koreader/koreader/blob/0b8d15416a3ebadee2b26a9d41af9c2047836d6e/frontend/document/documentregistry.lua#L146-L151

(It's quite possible the previous document is not yet closed at this point, as crengine may still delay the opening of the new one by saving the cache of the previous one, but switching from a big document, to a small one, and then between 2 small docs does not reclaim the large memory used by the first one.)

collectgarbage() does not help much with the memory usage rises I witness, I've already tried to set it at various places (and let some calls in the code just to be sure to get back what we can... which is not much).

Reply:
> That's a keyboard, right? Do we have a chip-like glyph available instead? That vaguely sounds more appropriate here, but I'm nitpicking ;).

@NiLuJe 
I like that icon. It look almost like chip :)